frame/message-queue/src/lib.rs

```diff
@@ -1085,6 +1090,100 @@ impl<T: Config> Pallet<T> {
 		ItemExecutionStatus::Executed(is_processed)
 	}
 
+	/// Ensure the correctness of state of this pallet.
+	///
+	/// # Assumptions-
+	///
+	/// If `serviceHead` points to a ready Queue, then BookState of that Queue has:
+	///
+	/// * `message_count` > 0
+	/// * `size` > 0
+	/// * `end` > begin
+	/// * Some(ready_neighbours)
+	/// * If `ready_neighbours.next` == self.origin, then `ready_neighbours.prev` == self.origin
+	///   (only queue in ring)
+	///
+	/// For Pages(begin to end-1) in BookState:
+	///
+	/// * `remaining` > 0
+	/// * `remaining_size` > 0
+	/// * `first` <= `last`
+	/// * Every page can be decoded into peek_* functions
+
+	pub fn do_try_state() -> Result<(), &'static str> {
+		// No state to check
+		if ServiceHead::<T>::get().is_none() {
+			return Ok(())
+		}
+
+		//loop around this origin
+		let starting_origin = ServiceHead::<T>::get().unwrap();
+
+		let mut service_head = ServiceHead::<T>::get();
+
+		loop {
```

Comment:
I dont really get the logic here. There is no `else`.  
```suggestion
		while let Some(head) = Self::bump_service_head(&mut WeightMeter::max_limit()) {
```
Now you just need to check if it is at `starting_origin` again. Note that the bump function returns the current one.